automl/amltk
# Add functionality to construct a PyTorch Model from a pipeline

This pull request introduces a script for building and evaluating a simple PyTorch model on the MNIST dataset. The script provides functionality for constructing the neural network model from a pipeline, training the model, and evaluating its performance.

> **_NOTE:_** The new functionality is not tied specifically to PyTorch. If we were to replace all instances of torch (e.g., `torch.nn.Linear`) in the pipeline with instances from another framework, the functionality would remain unchanged.

#### Changes Made:
- Added a new script `pytorch_example.py` for building and evaluating a PyTorch model on the MNIST dataset.
- Implemented classes `MatchDimensions` and `MatchChosenDimensions` to handle dimension matching between layers.
- Implemented `build_model_from_pipeline` function for building a PyTorch model from the provided pipeline.
- Added unit tests.
 
#### What does this implement/fix? Explain your changes:
This PR implements a new feature to build and evaluate a PyTorch model on the MNIST dataset. The script allows users to define a pipeline for constructing the model architecture, train the model with specified hyperparameters, and evaluate its accuracy.

#### Minimal Example / How should this PR be tested?
To test this PR, follow these steps:
1. Run the `pytorch_example.py` script.
2. Ensure that the PyTorch model is constructed and trained on the MNIST dataset.
3. Evaluate the model's accuracy on the test dataset.
4. Verify that the model performs satisfactorily according to the defined metrics.
